testsuite/cluster-test/src/experiments/packet_loss_random_validators.rs

```diff
@@ -0,0 +1,78 @@
+/// This module provides an experiment which introduces packet loss for
+/// a given number of instances in the cluster. It undoes the packet loss
+/// in the cluster after the given duration
+use crate::{
+    cluster::Cluster,
+    effects::{Action, PacketLoss, RemoveNetworkEffects},
+    experiments::Experiment,
+    instance::Instance,
+};
+use failure;
+use rand::Rng;
+use std::{collections::HashSet, fmt, thread, time::Duration};
+
+pub struct PacketLossRandomValidators {
+    instances: Vec<Instance>,
+    percent: f32,
+    duration: Duration,
+}
+
+impl PacketLossRandomValidators {
+    pub fn new(count: usize, percent: f32, duration: Duration, cluster: &Cluster) -> Self {
+        if count > cluster.instances().len() {
+            panic!(
+                "count {} should be <= number of instances {}",
+                count,
+                cluster.instances().len()
+            );
+        }
+        let mut instances = Vec::with_capacity(count);
+        let mut all_instances = cluster.instances().clone();
+        let mut rnd = rand::thread_rng();
+        for _i in 0..count {
```

Comment:
I added `Cluster::split_n_random` in #1531, do you mind switching to it at some point?

https://github.com/libra/libra/pull/1531/files#diff-3b9be419cfe902ff91ce042d9fb56865R145